odoo/tools/xml_utils.py

```diff
@@ -83,3 +88,49 @@ def create_xml_node(parent_node, node_name, node_value=None):
     :returns (etree._Element):
     """
     return create_xml_node_chain(parent_node, [node_name], node_value)[0]
+
+
+def load_xsd_from_url(env, url, module_name):
+    """Load XSD file or ZIP archive and save it as ir.attachment.
+    If the XSD file/archive has already been saved in database, then just return the attachment.
+
+    The ir.attachment file name is of the following format:
+
+    *<module_name>.xsd_cached_<short_url>*
+
+    Where *short_url* is the last part of the provided URL (each part is separated by a slash),
+    with every dot replaced by an underscore.
+
+    This format is used to ensure the uniqueness of the attachment, based on the module it is used for.
+
+    :param odoo.api.Environment env: environment of calling module
+    :param str url: URL of XSD file/ZIP archive
+    :param str module_name: name of calling module
+    :rtype: odoo.api.ir.attachment
+    :return: the attachment or False if an error occurred
+    """
+
+    if not url.endswith(('.xsd', '.zip')):
+        _logger.warning("The URL needs to lead to an XSD file or a ZIP archive")
+        return False
+
+    short_url = url.split('/')[-1].replace('.', '_')
+    xsd_file_name = '%s.xsd_cached_%s' % (module_name, short_url)
+
+    fetched_attachment = env['ir.attachment'].search([('name', '=', xsd_file_name)])
```

Comment:
Could be nice to have an additional helper to retrieve the attachment based on the same logic to avoid:
https://github.com/odoo/enterprise/pull/24195/files#diff-bc667f682f8a9ff34fb265bbe250a28f0e97efa9ab33fd949908ea33f4fc767cR63